packages/hooks/src/useRegisterPasskey.ts

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { useQuery, useMutation, useInvalidateQuery } from '@deriv/api';
+import { startRegistration } from '@simplewebauthn/browser';
+
+//TODO check error handling after deployment
+//TODO check the flow, refactor and add test cases when BE is ready
+const useRegisterPasskey = () => {
+    const invalidate = useInvalidateQuery();
+
+    const [device_registration_error, setRegistrationError] = React.useState('');
+    const [is_passkey_registered, setIsPasskeyRegistered] = React.useState(false);
+
+    const {
+        data,
+        refetch,
+        error: request_for_registration_error,
+        isFetching,
+    } = useQuery('passkeys_register_options', {
+        options: {
+            enabled: false,
```

Comment:
why do we set `enabled` to false?

Reply:
because we don't need to work with rendering. we trigger it manually with clicking 'refetch'